BlazarUI/app/scripts/components/shared/RepoBranchCardStack.jsx

```diff
@@ -0,0 +1,67 @@
+import React, {Component, PropTypes} from 'react';
+import Immutable from 'immutable';
+import {CardStack} from 'card-stack-test';
+
+
+import RepoBranchCard from './RepoBranchCard.jsx';
+import RepoBranchCardStackHeader from './RepoBranchCardStackHeader.jsx';
+import RepoBranchCardStackZeroState from './RepoBranchCardStackZeroState.jsx';
+
+class RepoBranchCardStack extends Component {
+
+  shouldComponentUpdate(nextProps) {
+    return JSON.stringify(this.props.starredBuilds.toJS()) !== JSON.stringify(nextProps.starredBuilds.toJS())
+      || this.props.expandedCard !== nextProps.expandedCard
+      || JSON.stringify(this.props.moduleBuildsList) !== JSON.stringify(nextProps.moduleBuildsList);
+  }
+
+  renderCards() {
+    const numberOfBuilds = this.props.starredBuilds.size;
+
+    return this.props.starredBuilds.map((build, key) => {
+      return (
+        <RepoBranchCard
+          moduleBuildsList={this.props.moduleBuildsList}
+          onClick={() => this.props.onClick(key, build)}
+          key={key}
+          expanded={key === this.props.expandedCard}
+          belowExpanded={key === this.props.expandedCard + 1 && this.props.expandedCard !== -1}
```

Comment:
The Card components should only have an expanded property so that we don't have to worry about manually computing and passing in the belowExpanded, first, and last properties. Instead of using the card-stack__card--first, card-stack__card--last, and card-stack__card--below-expanded classes, we could try to use some psuedo classes